### PR TITLE
Ship Placement Gui logic, except Validator. 

### PR DESCRIPTION
**Pull request by pawlusha**
observer pattern for gui event handling. 
Test for the above. 
TODO: validate ship position

### CODE REVIEW

File: src/main/java/com/fourmusketeers/client/View/TileButton.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import javafx.scene.control.Button;
+
+import java.util.Observable;
+
+/**
+ * @author Pawel_Ligeza
+ * @since 26.07.2017
+ */
+public class TileButton extends Observable {
+    private Button button;
```
Comment (Ice94) [inline src/main/java/com/fourmusketeers/client/View/TileButton.java]: Add empty line.

File: src/test/java/com/fourmusketeers/client/logic/ShipTest.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * @author Pawel_Ligeza
+ * @since 26.07.2017
+ */
+public class ShipTest {
+@Test
+public void shouldReturnTileIndicesList() {
+   // given
+    Ship ship = new Ship(4);
```
Comment (Ice94) [inline src/test/java/com/fourmusketeers/client/logic/ShipTest.java]: empty line

File: src/test/java/com/fourmusketeers/client/logic/ShipTest.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+   // when
+    ship.addIndexTile(2);
+    ship.addIndexTile(3);
+    ship.addIndexTile(4);
```
Comment (Ice94) [inline src/test/java/com/fourmusketeers/client/logic/ShipTest.java]: empty line.

File: src/test/java/com/fourmusketeers/client/logic/ShipsContainerTest.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import static org.testng.Assert.*;
+
+/**
+ * @author Pawel_Ligeza
+ * @since 26.07.2017
+ */
+public class ShipsContainerTest {
+
+    @Test
+    public void testInsertShip() throws Exception {
```
Comment (Ice94) [inline src/test/java/com/fourmusketeers/client/logic/ShipsContainerTest.java]: Rename for example ```shouldInsertGivenShip```.

File: src/test/java/com/fourmusketeers/client/logic/ShipsContainerTest.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        ShipsContainer shipsContainer = new ShipsContainer();
+        Ship ship = new Ship(4);
```
Comment (Ice94) [inline src/test/java/com/fourmusketeers/client/logic/ShipsContainerTest.java]: Magic number.

File: src/main/java/com/fourmusketeers/client/View/ShipPlacementWindowView.java [hunk 6]
```diff
@@ -38,10 +52,10 @@ void buildScene() {
     }
 
     private void addTileButtonsToGrid(GridPane gridPane) {
+        createTileButtons();
         for (int row = 0; row < Main.BOARD_SIZE; row++) {
             for (int col = 0; col < Main.BOARD_SIZE; col++) {
-                //TODO tileButton class
-                gridPane.add(new Button(), col, row);
+                gridPane.add(tileButtons.get(row*Main.BOARD_SIZE+col).getButton(), col, row);
```
Comment (Devas) [inline src/main/java/com/fourmusketeers/client/View/ShipPlacementWindowView.java]: Autoformat

File: src/main/java/com/fourmusketeers/client/View/ShipPlacementWindowView.java [hunk 7]
```diff
@@ -51,4 +65,16 @@ private Button createNextButton() {
         nextButton.setOnAction(event -> new GameWindowView(primaryStage, this).display());
         return nextButton;
     }
+
+    @Override
+    public void update(Observable o, Object arg) {
```
Comment (Devas) [inline src/main/java/com/fourmusketeers/client/View/ShipPlacementWindowView.java]: Change names

File: src/main/java/com/fourmusketeers/client/logic/ShipsGenerator.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+    ShipsGenerator(ShipsContainer shipsContainer) {
+        this.shipsContainer = shipsContainer;
+        this.shipValidator = new ShipValidator();
+        mastSizeIterator = IntStream.of(4, 3, 3, 2, 2, 2, 1, 1, 1, 1).boxed().iterator();
+        currentShip = new Ship(mastSizeIterator.next());
+    }
+
+    @Override
+    public void update(Observable o, Object arg) {
```
Comment (Devas) [inline src/main/java/com/fourmusketeers/client/logic/ShipsGenerator.java]: Rename

File: src/test/java/com/fourmusketeers/client/logic/ShipTest.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.*;
+
+/**
+ * @author Pawel_Ligeza
+ * @since 26.07.2017
+ */
+public class ShipTest {
+    @Test
```
Comment (Devas) [inline src/test/java/com/fourmusketeers/client/logic/ShipTest.java]: Enter

**Pull request merged**